UnattendResources/Logon.ps1

```diff
@@ -75,9 +75,16 @@ function Clean-UpdateResources {
     Remove-ItemProperty -Path "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Winlogon" -Name AutoLogonCount -ErrorAction SilentlyContinue
 
     # Cleanup
-    Remove-Item -Recurse -Force $resourcesDir
+    Remove-Item -Recurse -Force -Exclude "sdelete.exe" $resourcesDir
     Remove-Item -Force "$ENV:SystemDrive\Unattend.xml"
-
+    $sdelete = "$resourcesDir\sdelete.exe"
+    if ( Test-Path $sdelete ) {
+        Write-Host "Optimizing for sparse image..."
+        & $sdelete -AcceptEULA -z $ENV:SystemDrive

```

Comment:
I think here it would be nice to have a LASTEXITCODE check maybe?

Reply:
I'm not shure, if sdelete probably uses some special exit codes. Will check its source.

Reply:
Regarding sdelete.exe accidentially not existing:

As far as I understand, Logon.ps1 should not be stopped on error anyway. So the only option that makes sense is to print a warning.

Reply:
I checked the source of sdelete and came to the conclusion that a LASTEXITCODE check is unnecessary, because:

1. sdelete prints its own error messages
2. we would only be able to print a warning anyway

Reply:
Got it, will do some runs and make sure the sdelete cannot break the generation also on older versions like win7.

Reply:
sdelete itself is ancient. It should work on any old WIndows. Quote from the source:
```
// SDelete - Secure Delete
// Copyright (C) 1999 Mark Russinovich
// Systems Internals - http://www.sysinternals.com
```

Reply:
as @alexpilotti said, this part can be left as it is. if sdelete is present, it will use it.